packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/delete/delete.routing.js

```diff
@@ -1,37 +1,78 @@
+import kebabCase from 'lodash/kebabCase';
+import { BillingService } from '@ovh-ux/manager-models';
+
 export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('app.account.billing.autorenew.delete', {
     url: '/delete?serviceId&serviceType',
-    views: {
-      modal: {
-        component: 'billingAutorenewDelete',
-      },
-    },
-    layout: 'modal',
-    translations: { value: ['.'], format: 'json' },
+    componentProvider: /* @ngInject */ (engagement) =>
+      engagement ? 'billingAutorenewDelete' : 'billingConfirmTermination',
     resolve: {
       cancelResiliationUrl: /* @ngInject */ ($state, serviceId) =>
         $state.href('app.account.billing.autorenew.cancelResiliation', {
           serviceId,
         }),
       engagement: /* @ngInject */ (Server, service) =>
         (service.canHaveEngagement()
-          ? Server.getSelected(service.serviceId)
+          ? Server.getSelected(service.domain)
           : Promise.resolve({ engagement: null })
         ).then(({ engagement }) => engagement),
-      goBack: /* @ngInject */ (goToAutorenew) => goToAutorenew,
+
+      goBack: /* @ngInject */ (
+        $translate,
+        cancelResiliationUrl,
+        goToAutorenew,
+      ) => (success = false) =>
+        success
+          ? goToAutorenew(
+              `${$translate.instant('autorenew_service_delete_success')}
+      <a data-href="${cancelResiliationUrl}" data-translate="autorenew_service_delete_cancel"></a>`,
+            )
+          : goToAutorenew(),
       serviceId: /* @ngInject */ ($transition$) =>
         $transition$.params().serviceId,
       serviceType: /* @ngInject */ ($transition$) =>
         $transition$.params().serviceType,
       service: /* @ngInject */ (BillingAutoRenew, serviceId, serviceType) =>
-        BillingAutoRenew.getService(serviceId, serviceType),
+        BillingAutoRenew.getService(serviceId, serviceType).then(
+          (service) =>
+            new BillingService({
+              ...service,
+              serviceId: service.id,
+            }),
+        ),
       supportPhoneNumber: /* @ngInject */ (constants, currentUser) =>
         constants.SUPPORT[currentUser.ovhSubsidiary],
-      updateService: /* @ngInject */ (BillingAutoRenew) => (service) =>
-        BillingAutoRenew.updateService(service),
+      confirmTermination: /* @ngInject */ (
+        atInternet,
+        BillingAutoRenew,
+        service,
+      ) => () => {
+        atInternet.trackClick({
+          name: `autorenew::${kebabCase(service.serviceType)}::delete::confirm`,
+          type: 'action',
+          chapter1: 'dedicated',
+          chapter2: 'account',
+          chapter3: 'billing',
+        });
+        service.setForResiliation();
```

Comment:
`setFormResiliation` ?

Reply:
No `setForResiliation` is a method on the `BillingService` class to set all `renew` fields correctly for resilitation